src/core/sock/sockinfo_tcp.cpp

```diff
@@ -6046,3 +6047,204 @@ inline bool sockinfo_tcp::handle_bind_no_port(int &bind_ret, in_port_t in_port,
 
     return CONTINUE_WITH_BIND;
 }
+
+int sockinfo_tcp::tcp_tx_express(const struct iovec *iov, unsigned iov_len, uint32_t mkey,
+                                 xlio_express_flags flags, void *opaque_op)
+{
+    if (unlikely(!is_rts())) {
```

Comment:
Why not use is_connected_and_ready_to_send() with param if to do wait_rx. We better have a single point of truce.

Reply:
I'll explore it

Reply:
It might be more correct here to use the is_connected_and_ready_to_send the same as in regular tcp_tx.
```
if (unlikely(!is_connected_and_ready_to_send())) {                                                                                                                                                                                                                                                                                  
    return -1;
}
```